### ISSUE TITLE
Improvement request: add support for Guava immutable enum map / set for rule OptimizeMapOrSetForEnum

### ISSUE THREAD

**Issue by ntinnemeier** [jborgers/PMD-jPinpoint-rules#167]
The OptimizeMapOrSetForEnum rule triggers on the following code, while under the hood we do use an immutable enum map:

```
public class GreetingsGuavaViolation {
    enum Language {
        NL, EN, FR;
    }

    private static final Map<Language, String> greetings = Maps.immutableEnumMap(
            ImmutableMap.of(
                NL, "hallo",
                EN, "hello",
                FR, "bonjour"));
}
```

The difficulty here is that `Maps.immutableEnumMap()` returns an `ImmutableEnumMap` which is an `ImmutableMap`, but not an `EnumMap`. That is to say, the following line of code won't compile:
```
private static final EnumMap<Language, String> greetings = Maps.immutableEnumMap(...)
```
Within Guava the class `ImmutableEnumMap` is not a public class, so we also can't write:
```
private static final ImmutableEnumMap<Language, String> greetings = Maps.immutableEnumMap(...)
```

This is issue is related to (and might be fixed by): https://github.com/jborgers/PMD-jPinpoint-rules/issues/166
Without https://github.com/jborgers/PMD-jPinpoint-rules/issues/166 being fixed, I can't test if Guava's `ImmutableEnumMap` (and `ImmutableEnumSet`) is actually supported.

**Comment by jborgers** [comment]
Rule does not give a violation on the test case code anymore.
